src/Token.php

```diff
@@ -252,17 +252,16 @@ public function extract()
                 if ($this->flags & self::FLAG_NUMBER_HEX) {
                     if ($this->flags & self::FLAG_NUMBER_NEGATIVE) {
                         $ret = str_replace('-', '', $this->token);
-                        sscanf($ret, '%x', $ret);
-                        $ret = -$ret;
+                        $ret = -hexdec($ret);
```

Comment:
```suggestion
                        $ret -= hexdec($ret);
```
Would this be more obvious?

Reply:
I don't think it's the same behavior. If, for example, `$this->token` worth `-a0` (-16 in decimal), expected value for `$ret` is -16. If you do `$ret -= hexdec($ret)`, you're doing `$ret = a0 - 16`, so basically, `$ret = 0`.

Reply:
Yes, I did not read it right
Can you add a test so we are sure nobody removes the sign by mistake?